### PR TITLE
Fix console errors and warnings in Data Processing and Data Exploration

### PR DESCRIPTION
**Pull request by ivababukova**
hedgehog
# Background
#### Link to issue 

https://biomage.atlassian.net/browse/BIOMAGE-489

#### Link to staging deployment URL 

Will be available here: https://ui-ivas-pr.scp-staging.biomage.net/ shortly

#### Links to any Pull Requests related to this
N/A

#### Anything else the reviewers should know about the changes here
Console errors in plots and tables page are not looked at.

# Changes
### Code changes
Mostly fixes of invalid prop types and missing keys.

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written N/A
- [x] Tested locally with Inframock 
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [x] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/components/plots/FrequencyPlot.jsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ const FrequencyPlot = (props) => {
     hierarchy, properties, config, actions,
   } = props;
 
+  console.log('***** ', hierarchy);
```
Comment (aerlaut) [inline src/components/plots/FrequencyPlot.jsx:13]: Don't forget to remove `console.log`

File: src/components/plots/ElbowPlot.jsx [hunk 2]
```diff
@@ -131,8 +131,8 @@ ElbowPlot.defaultProps = {
 ElbowPlot.propTypes = {
   stylingConfig: PropTypes.object.isRequired,
   calculationConfig: PropTypes.object.isRequired,
-  plotData: PropTypes.object.isRequired,
-  actions: PropTypes.object.isRequired,
+  plotData: PropTypes.array.isRequired,
+  actions: PropTypes.bool.isRequired,
```
Comment (aerlaut) [inline src/components/plots/ElbowPlot.jsx]: Actions can actually be boolean or object : `https://github.com/vega/vega-embed`. Maybe you can use `PropTypes.oneOfType`?

```
  // An object that could be one of many types
  actions: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.object,
  ])
```

File: src/components/plots/FrequencyPlot.jsx (hunk 1 above)
Comment (aerlaut) [inline src/components/plots/FrequencyPlot.jsx]: Don't forget to remove `console.log`

File: src/components/plots/FrequencyPlot.jsx [hunk 3]
```diff
@@ -23,10 +24,10 @@ FrequencyPlot.defaultProps = {
 };
 
 FrequencyPlot.propTypes = {
-  hierarchy: PropTypes.object.isRequired,
+  hierarchy: PropTypes.array.isRequired,
   properties: PropTypes.object.isRequired,
   config: PropTypes.object.isRequired,
-  actions: PropTypes.object,
+  actions: PropTypes.bool,
```
Comment (aerlaut) [inline src/components/plots/FrequencyPlot.jsx]: Actions can actually be boolean or object : `https://github.com/vega/vega-embed`. Maybe you can use `PropTypes.oneOfType`?

File: src/pages/_app.jsx [hunk 4]
```diff
@@ -103,8 +103,8 @@ const WrappedApp = ({ Component, pageProps }) => {
 };
 
 WrappedApp.propTypes = {
-  Component: PropTypes.object.isRequired,
-  pageProps: PropTypes.string.isRequired,
+  Component: PropTypes.func.isRequired,
```
Comment (aerlaut) [inline src/pages/_app.jsx]: is `PropTypes.element` more suitable?

File: src/components/plots/ElbowPlot.jsx (hunk 2 above)
Comment (ivababukova) [reply src/components/plots/ElbowPlot.jsx]: yup, that works too. I did not see the object type used anywhere in the code though

File: src/pages/_app.jsx (hunk 4 above)
Comment (ivababukova) [reply src/pages/_app.jsx]: it has to be a func, otherwise we get a warning

File: src/pages/experiments/[experimentId]/data-processing/index.jsx [hunk 5]
```diff
@@ -57,7 +57,7 @@ const DataProcessingPage = ({ experimentId, experimentData, route }) => {
     {
       key: 'dataIntegration',
       name: 'Data integration',
-      render: (key) => <DataIntegration filtering key={key} />,
```
Comment (ivababukova) [inline src/pages/experiments/[experimentId]/data-processing/index.jsx:60]: what is this `filtering` parameter used for in this page? I saw it in a couple of other places, but I didn't see where it is getting set from

File: src/components/plots/ElbowPlot.jsx (hunk 2 above)
Comment (aerlaut) [reply src/components/plots/ElbowPlot.jsx]: Yup, I don't think it's currently used yet.

File: src/pages/experiments/[experimentId]/data-processing/index.jsx (hunk 5 above)
Comment (aerlaut) [reply src/pages/experiments/[experimentId]/data-processing/index.jsx:60]: the `filtering` key is not used, to it should be removed

Comment (ivababukova) [reply src/pages/experiments/[experimentId]/data-processing/index.jsx:60]: can I remove it in the other filters too?

Comment (ivababukova) [reply src/pages/experiments/[experimentId]/data-processing/index.jsx:60]: no point having things that are not used

Comment (aerlaut) [reply src/pages/experiments/[experimentId]/data-processing/index.jsx:60]: From what I see, they were used as a flag to enable / disable one of the plot styling configs in the old data processing components. I think you can leave them be at the moment. They should be removed later on as the components are being worked on,

**Review by aerlaut — APPROVED** [review]
Solved in comments. `filtering` in old Data Processing components can be left there. They should be replaced as the component is worked on.

**Pull request merged**